### PR TITLE
Husky for link-checking staged only files

### PR DESCRIPTION
**Pull request by lorenaramonda**
I saw that the pre-commit hook in package.json was looking for lint errors in all the code of the commit, even that one unstaged, I re-wrote the rule using husky and telling to check only the real code that it's about to be committed (staged only).

I think it's a better way, because sometimes you want to keep something unstaged in your local to use after the commit (maybe for personal tests or whatever) and it's non-sense that the hook prevents you to commit if there are lint errors in unstaged code.

What do you think about it?

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added test:unit:watch with a workaround of a jest problem with template strings - @resubaka (#3351, #3354)
 - Added test to helpers/index.ts so it is partly tested - @resubaka (#3376, 3377)
 - Added config for the defaultTitle compitable with multistore - @cnviradiya (#3282)
+- Added husky package to manage lint check only for staged files in git
```
Comment (ResuBaka) [inline CHANGELOG.md]: Can you add your PR number and @ tag to the entry.

Comment (lorenaramonda) [reply CHANGELOG.md]: yes sure 👍

Comment (lorenaramonda) [reply CHANGELOG.md]: done!

Comment (ResuBaka) [reply CHANGELOG.md]: Thanks.

**Comment by lorenaramonda** [comment]
> Cool @lorenaramonda ! Could you with this occasion add running `test:unit` for `pre-push`?

Hi @patzick I think it's better to move this into a separate issue because it's not as easy as it seems.
Adding just: `"pre-push": "yarn test:unit"` throws an error about some import @babel/polyfill and I think it's better solving it separately not to keep this PR blocked.
What do u think about it?

**Comment by pkarw** [comment]
@lorenaramonda I agree wth you

**Comment by patzick** [comment]
Thanks, right call! I'll add an issue for that:)

**Pull request merged**